src/features/Students/data/slices.js

```diff
@@ -0,0 +1,69 @@
+/* eslint-disable react/jsx-filename-extension */
+import React, {
+  createContext,
+  useContext,
+  useReducer,
+  useMemo,
+} from 'react';
+import PropTypes from 'prop-types';
+
+const StudentEnrollmentsContext = createContext();
+
+const initialState = {
+  data: [],
+  status: 'success',
+  error: null,
+  itemsPerPage: 10,
+  currentPage: 1,
+  numPages: 0,
+};
+
+const reducer = (state, action) => {
+  switch (action.type) {
+    case 'FETCH_REQUEST':
+      return { ...state, status: 'loading' };
+    case 'FETCH_SUCCESS': {
+      const { results, count } = action.payload;
+      const numPages = Math.ceil(count / state.itemsPerPage);
+      return {
+        ...state,
+        status: 'success',
+        data: results,
+        numPages,
+        count,
+      };
+    }
+    case 'FETCH_FAILURE':
+      return { ...state, status: 'error', error: action.payload };
+    case 'UPDATE_CURRENT_PAGE':
+      return { ...state, currentPage: action.payload };
+    default:
+      return state;
+  }
+};
+
+const StudentEnrollmentsProvider = ({ children }) => {
+  const [state, dispatch] = useReducer(reducer, initialState);
+  const contextValue = useMemo(() => ({ state, dispatch }), [state, dispatch]);
+
+  return (
+    <StudentEnrollmentsContext.Provider value={contextValue}>
+      {children}
+    </StudentEnrollmentsContext.Provider>
+  );
+};
+
+const useStudentEnrollments = () => {
+  const context = useContext(StudentEnrollmentsContext);
+
+  if (!context) {
+    throw new Error('useStudentEnrollments must be used within a StudentEnrollmentsProvider');
+  }
```

Comment:
```suggestion
  };
```

Reply:
@anfbermudezme unncessery semi colon according to Lint.